libp2p/peer/peerstore.py

```diff
@@ -10,90 +12,196 @@
+        if peer_id in self.peer_data_map:
+            return self.peer_data_map[peer_id]
         data = PeerData()
-        self.peer_map[peer_id] = data
-        return self.peer_map[peer_id]
+        self.peer_data_map[peer_id] = data
+        return self.peer_data_map[peer_id]
 
     def peer_info(self, peer_id: ID) -> Optional[PeerInfo]:
-        if peer_id in self.peer_map:
-            peer_data = self.peer_map[peer_id]
+        """
+        :param peer_id: peer ID to get info for
+        :return: peer info object
+        """
+        if peer_id in self.peer_data_map:
+            peer_data = self.peer_data_map[peer_id]
             return PeerInfo(peer_id, peer_data.addrs)
         return None
 
     def get_protocols(self, peer_id: ID) -> List[str]:
-        if peer_id in self.peer_map:
-            return self.peer_map[peer_id].get_protocols()
+        """
+        :param peer_id: peer ID to get protocols for
+        :return: protocols (as list of strings)
+        :raise PeerStoreError: if peer ID not found
+        """
+        if peer_id in self.peer_data_map:
+            return self.peer_data_map[peer_id].get_protocols()
         raise PeerStoreError("peer ID not found")
 
     def add_protocols(self, peer_id: ID, protocols: Sequence[str]) -> None:
+        """
+        :param peer_id: peer ID to add protocols for
+        :param protocols: protocols to add
+        """
         peer = self.__create_or_get_peer(peer_id)
         peer.add_protocols(list(protocols))
 
     def set_protocols(self, peer_id: ID, protocols: Sequence[str]) -> None:
+        """
+        :param peer_id: peer ID to set protocols for
+        :param protocols: protocols to set
+        """
         peer = self.__create_or_get_peer(peer_id)
         peer.set_protocols(list(protocols))
 
     def peer_ids(self) -> List[ID]:
-        return list(self.peer_map.keys())
+        """
+        :return: all of the peer IDs stored in peer store
+        """
+        return list(self.peer_data_map.keys())
 
     def get(self, peer_id: ID, key: str) -> Any:
-        if peer_id in self.peer_map:
+        """
+        :param peer_id: peer ID to get peer data for
+        :param key: the key to search value for
+        :return: value corresponding to the key
+        :raise PeerStoreError: if peer ID or value not found
+        """
+        if peer_id in self.peer_data_map:
             try:
-                val = self.peer_map[peer_id].get_metadata(key)
+                val = self.peer_data_map[peer_id].get_metadata(key)
             except PeerDataError as error:
                 raise PeerStoreError(error)
             return val
         raise PeerStoreError("peer ID not found")
 
     def put(self, peer_id: ID, key: str, val: Any) -> None:
+        """
+        :param peer_id: peer ID to put peer data for
+        :param key:
+        :param value:
+        """
         # <<?>>
         # This can output an error, not sure what the possible errors are
```

Comment:
anything to resolve here before merge?

Reply:
Seems like an outdated comment and `__create_or_get_peer` is replaced by `defaultdict` so it should be safe now.